### PR TITLE
fix bug: bad replacement if scm attribute was the last one in ClassDef

### PR DESCRIPTION
**Pull request by jgsogo**
Will fix failing test in #3505 
- [x] Add tests (see comment below)

### CODE REVIEW

**Comment by jgsogo** [comment]
Anyway, I will consider using a more powerful library that allows to print back the AST into a file...

**Comment by jgsogo** [comment]
There are three tests related to this that should be added:

- conanfile with scm and more attributes/functions (there are lots of this kind in the codebase) ✅ 
- conanfile with scm being the last element in the class (the one introduced in #3505) ✅ 
- conanfile with scm being the last element in the class, but with another class or global function below it. <-- maybe, at this moment, there isn't one example of this.

File: conans/test/client/cmd/export_test.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+
```
Comment (memsharded) [inline conans/test/client/cmd/export_test.py]: We are adding command tests to ``command`` test subfolder.

Comment (jgsogo) [reply conans/test/client/cmd/export_test.py]: Please, suggest a destination for this TestCase, `conans/test/command/unittest/export_test.py`?

**Comment by jgsogo** [comment]
Tests are complete. 

Side-note: when I think that there is task related to the PR that has to be done before merging, I create a checkbox in the description.

File: conans/client/cmd/export.py [hunk 2]
```diff
@@ -100,14 +100,20 @@ def _replace_scm_data_in_conanfile(conanfile_path, scm_data):
...
-    for item in tree.body:
+    for i_body, item in enumerate(tree.body):
         if isinstance(item, ast.ClassDef):
             statements = item.body
             for i, stmt in enumerate(item.body):
                 if isinstance(stmt, ast.Assign) and len(stmt.targets) == 1:
                     if isinstance(stmt.targets[0], ast.Name) and stmt.targets[0].id == "scm":
                         try:
-                            next_line = statements[i+1].lineno - 1
+                            if i+1 == len(statements):  # Last statement in my ClassDef
```
Comment (lasote) [inline conans/client/cmd/export.py]: i[space]+[space]1

**Comment by lasote** [comment]
I think with these new lines (I'm sure they are necessary, no complains) we reached the limit of ugly code. I would now prefer a shorter/cleaner fix even if it requires to introduce a new dependency

**Comment by lasote** [comment]
Merging this and opening a new issue for 1.9 to look for a better implementation using an external library.

**Pull request merged**